homeassistant/components/sma/__init__.py

```diff
@@ -143,7 +145,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     try:
         # Get updated device info
-        device_info = await sma.device_info()
+        sma_device_info = await sma.device_info()
+        device_info = DeviceInfo(
+            configuration_url=url,
+            default_manufacturer=DEFAULT_MANUFACTURER,
+            identifiers={(DOMAIN, entry.unique_id)},
+            manufacturer=sma_device_info.get("manufacturer"),
+            model=sma_device_info.get("type"),
+            name=sma_device_info.get("name"),
+            sw_version=sma_device_info.get("sw_version"),
```

Comment:
This was previously 
```
            manufacturer=self._device_info["manufacturer"],
            model=self._device_info["type"],
            name=self._device_info["name"],
            sw_version=self._device_info["sw_version"],
```

Can these values actually be missing?

If not we should use `[]` instead of `get`

Reply:
Yes you are right! Checking the `pysma` code I also noticed it will also provide a default value if no value was found on the device, so I removed `DEFAULT_MANUFACTURER` as well